lib/widgets/flame_animation_widget.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flutter/material.dart' hide Animation;
+import 'package:flame/animation.dart';
+
+import './flame_sprite_widget.dart';
+
+import 'dart:math';
+
+class FlameAnimationWidget extends StatefulWidget {
+  final Animation animation;
+  final bool center;
+  final bool play;
+
+  FlameAnimationWidget({
+    this.animation,
+    this.play = true,
+    this.center = false,
+  }) : assert(animation.loaded(), 'Animation must be loaded');
```

Comment:
is this restriction necessary? is that because you can't get a hook into when the animation does load to setState in the component?

this goes a bit against our standard of let it build & render ASAP but if it's a hassle to implement I'm 100% ok with it (just want to understand the reasoning)

Reply:
It sure is possible to make a way to handle this and somehow hook ask to rerender when it is loaded.

But when the matter are widgets, I think it is better to handle the loading state to the user, as they may want to display different widgets (like a `Text('Loading')` for example), I don't like the idea of this widget assuming that kind of stuff, so I made the assert to be sure that the component will be rendered.

And since Flutter has widgets like `FutureBuilder` it would be really easy to wrap this into a loader, and handle the states and so on.

Reply:
that makes sense to me.

Reply:
oh but maybe let's add a quick line about that in the docs? since it might be unexpected